src/utils.cpp

```diff
@@ -355,9 +355,51 @@ std::string utils::retrieve_url(const std::string& url,
 		curl_easy_setopt(easyhandle, CURLOPT_USERPWD, authinfo.c_str());
 	}
 
-	curl_easy_perform(easyhandle);
+	// Error handling as per https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html
+	char errbuf[CURL_ERROR_SIZE];
+	// Please note that we clobber CURLOPT_ERRORBUFFER here in case of cached handles
+	// Currently, this is the only place we do this, so this should be fine.
+	// There does not seem to be a way to query curlopts, so we cannot save the old errorbuf value here...
+	curl_easy_setopt(easyhandle, CURLOPT_ERRORBUFFER, errbuf);
+	errbuf[0] = '\0';
+
+	CURLcode res = curl_easy_perform(easyhandle);
+
+	// Report curl errors
+	if (res != CURLE_OK) {
+		std::string errmsg(errbuf);
+		if (errmsg.empty()) {
+			errmsg = curl_easy_strerror(res);
+		} else {
+			if (*errmsg.crbegin() != '\n') {
+				// Prettify: end logmessage with newline if not already done by libcurl
+				errmsg += "\n";
+			}
+		}
+
+		if (body != nullptr) {
```

Comment:
I see this was copied from below. Why not move it into a lambda or something, and call that at both places?

You can also take an opportunity to fix the `else` branch which doesn't mention the method :)

Reply:
In theory yes. However, both loglevel and argument and different parameter counts and types. So either it will be a lambda with varargs, or constructing segments of the output string beforehand by hand. All those options -- at least to me -- seemed like adding more line noise than actually being helpful. But I will try to come up with a proposal.